### ISSUE TITLE
False positive for missing maxLength for format: date

### ISSUE THREAD

**Issue by DavidBiesack** [stoplightio/spectral-owasp-ruleset#41]
When a property has

```
type: string
format: date
```

we get a false positive 

```
error  owasp:api4:2019-string-limit             Schema of type string must specify maxLength, enum, or const.
```

## Context

The `date` format must match `full-date       = date-fullyear "-" date-month "-" date-mday` 
as per RFC 3339 and ISO 8601, such as `2023-04-03` and thus there is an implicit
`minLength` and `maxLength` of 10 for this format, so there should be no error.

## Current Behavior

error diagnostic

```
error  owasp:api4:2019-string-limit             Schema of type string must specify maxLength, enum, or const.
```

## Expected Behavior

no error

## Possible Workaround/Solution

one can explicitly add

```
minLength: 10
maxLength: 10
```

but that  bloats the OpenAPI definition.

Note that the format may be defined via an `allOf` construct as well, so it may not be in the same JSON Schema object
as the `type: string`. This may be harder for Spectral to handle:

```
allOf:
  - $ref: '#/components/schemas/schemaWith_type:_string'
  - $ref: '#/components/schemas/schemaWith_format:_date'
```

## Environment

- Version used: spectral-owasp-ruleset 1.4.1 ; stoplight/spectral-cli 6.6.0
- Environment name and version (e.g. Chrome 39, node.js 5.4):
- Operating System and version (desktop or mobile): macOS

**Comment by esatterwhite** [comment]
Ya, I also get a false positive on valid string properties. It works correctly in 1.4.0, but not 1.4.0. But it doesn't look like the rule changed

![image](https://user-images.githubusercontent.com/148561/230516514-739da4de-0dd4-4506-9b4f-40031329c014.png)

In this case I've specified both enum and maxLength and its still giving me errors

**Comment by paul-kraftlauget** [comment]
I can also note that it is the same with 
```
type: string
format: date-time
```

For date-time, it is implicit that `maxLength: 30`.

I can also note that when generating code using open-api generators, at least with Java, if maxLength is added to a date or a date-time format, the code generator adds a @Size annotation which fails since java validation cannot calculate Size on a date or datetime java objects. So the workaround doesn't work entirely as expected for other parts of the application lifecycle.

This is similar for the UUID format as well:
https://github.com/OpenAPITools/openapi-generator/issues/9256

minLength and maxLength should not be required for these known formats.

**Comment by philsturgeon** [comment]
Seeing as format is by default merely an annotation it may not be used at all in the validaiton process. In order to keep things secure, if you're using OpenAPI-based validation, or if you just want to be clear about how that API is going to work, it's better to define the maxLength properties even when there is a format in place.

It might feel weird, but tbh format is weird, and there's no nice answer to that other than a time machine and/or making it never exist.

**Comment by gaetano23061984** [comment]
Hi all,
any news about the fixing of this false positive?

I'm already using 2.0.0 version.
